boards/raspberry_pi_pico/src/main.rs

```diff
@@ -388,18 +395,44 @@ pub unsafe fn main() {
             .finalize(());
     let _ = process_console.start();
 
+    let i2c0 = &peripherals.i2c0;
+    i2c0.init(10 * 1000, &peripherals.clocks, &peripherals.resets);
+
+    let sda_pin = peripherals.pins.get_pin(RPGpio::GPIO4);
+    let scl_pin = peripherals.pins.get_pin(RPGpio::GPIO5);
+
+    sda_pin.set_function(GpioFunction::I2C);
+    scl_pin.set_function(GpioFunction::I2C);
+
+    sda_pin.set_floating_state(FloatingState::PullUp);
+    scl_pin.set_floating_state(FloatingState::PullUp);
+
+    let i2c = static_init!(
+        I2CMasterDriver<I2c>,
+        I2CMasterDriver::new(
+            i2c0,
+            &mut capsules::i2c_master::BUF,
+            board_kernel.create_grant(
+                capsules::i2c_master::DRIVER_NUM,
+                &memory_allocation_capability
+            ),
+        )
+    );
+    peripherals.i2c0.set_master_client(i2c);
```

Comment:
No there is no explicit reason. I was just not aware that the I2C component serves as an alternative to this, but will look into it, thanks.

Reply:
I have now looked at I2C component, but can't really seem how it would work here. It seems to be intended for simplifying the definition of modules that are I2C clients, but the intention of this code was so that you can use i2c_master in a libtock-c app. Am I missing something?

Reply:
I don't think there is a component for that yet.